app/controllers/schools/courses_controller.rb

```diff
@@ -223,6 +223,20 @@ def evaluation_criteria
         authorize(scope.find(params[:id]), policy_class: Schools::CoursePolicy)
     end
 
+    # GET /courses/:id/clone
+    def clone
+      course = authorize(
+        scope
+          .where(id: params[:id])
+          .includes(:evaluation_criteria, :levels, :target_groups, :targets)
+          .first,
+        policy_class: Schools::CoursePolicy
+      )
+      new_name = [course.name, "copy"].join(" - ")
+      ::Courses::CloneService.new(course).clone(new_name, course.school)
```

Comment:
The cloning operation is currently performed inline 
with the request. The operation could take a long time, and so should be deferred using ActiveJob. Ideally, this operation is handled using a mutation (since that opens up the possibility for API calls), but a POST request that queues using an ActiveJob is also fine.

Deferring also means the preloading in this action isn't required.

Reply:
yeah... right. I will do it async and with mutation.